### DISCUSSION TITLE
(Python) Does renovate pin build dependencies?

### DISCUSSION THREAD

**Discussion by hasansalimkanmaz** [renovatebot/renovate#15987] [Request Help] [answered]
In python, build dependencies are the dependencies required by the dependencies of a project. The build dependencies can be unpinned depending on the package installs it. As a result of this, build dependencies can be breaking point for certain codebases even if the dependencies of the codebase are pinned, when a buggy release is just released for the build dependency.

To solve the above issue, one simple idea is also pinning build dependencies, however, it is not so easy to implement as there will be lots of build dependencies in a codebase behind the scenes and collecting all of them in a `requirements.txt` file results in a messy/unclear `requirements.txt` file.

In the documentation, I see how renovate works as follows:

>  How it works[¶](https://docs.renovatebot.com/configuration-options/#how-it-works)
> 
> Renovate searches through each repository for package files
> Existing dependencies are extracted from the package files
> Renovate looks up the latest version on [PyPI](https://pypi.org/) to determine if any upgrades are available
> If the source package includes a GitHub URL as its source, and has a "changelog" file or uses GitHub releases, a Release Note will be embedded in the generated PR

However, it is still unclear for me whether renovate manages build dependencies or not. Does renovate searches the dependency files to find build dependencies?

**Comment by viceice** [comment, marked as answer]
use `pip-compile`  manager

- https://docs.renovatebot.com/modules/manager/pip-compile/
- https://github.com/renovatebot/renovatebot.github.io/blob/main/requirements.in

**Reply by hasansalimkanmaz** [reply]
Thanks for the rapid reply and sorry for missing the documentations. I didn't know what `pip-compile` is before :) . It is the solution that I am looking for.
